internal/controllers/topology/cluster/conditions.go

```diff
@@ -139,6 +140,32 @@ func (r *Reconciler) reconcileTopologyReconciledCondition(s *scope.Scope, cluste
 
 	// If any of the lifecycle hooks are blocking any part of the reconciliation then topology
 	// is not considered as fully reconciled.
+	if feature.Gates.Enabled(feature.RuntimeSDK) {
+		if annotations.HasWithPrefix(clusterv1.BeforeClusterUpgradeHookAnnotationPrefix, cluster.Annotations) {
```

Comment:
I think this is not correct. 

We assume that an upgrade is blocked right now, just because the annotations exist.

The annotations can exist at any point in the cluster lifecycle. This doesn't mean that they block anything.

I would recommend storing some info in the s.HookResponseTracker in desired state and then using it here

(Ideally in a way that we don't have to duplicate the logic in the for loop into both places)

Reply:
Thinking a bit about it. Wondering if we can even reuse l.169++ (if there's an elegant way to do it)

I would try to reuse AggregateMessage but have something additional and not just try to use AggregateRetryAfter